shared-module/rgbmatrix/RGBMatrix.c

```diff
@@ -84,31 +84,36 @@ void common_hal_rgbmatrix_rgbmatrix_reconstruct(rgbmatrix_rgbmatrix_obj_t* self,
         _PM_FREE(self->core.screenData);
 
         self->framebuffer = NULL;
-        self->bufinfo.buf = _PM_allocator_impl(self->bufsize);
+        self->bufinfo.buf = common_hal_rgbmatrix_allocator_impl(self->bufsize);
         self->bufinfo.len = self->bufsize;
         self->bufinfo.typecode = 'H' | MP_OBJ_ARRAY_TYPECODE_FLAG_RW;
     }
 
+    memset(&self->core, 0, sizeof(self->core));
     ProtomatterStatus stat = _PM_init(&self->core,
         self->width, self->bit_depth,
         self->rgb_count/6, self->rgb_pins,
         self->addr_count, self->addr_pins,
         self->clock_pin, self->latch_pin, self->oe_pin,
         self->doublebuffer, self->timer);
 
+    common_hal_rgbmatrix_rgbmatrix_deinit(self);
```

Comment:
Why always deinit?

Reply:
This line is wrong and should be deleted.